rust/src/lib/ip.rs

```diff
@@ -147,6 +149,14 @@ pub struct InterfaceIpv4 {
     /// which result the IP addresses after first one holding the `secondary`
     /// flag.
     pub addresses: Option<Vec<InterfaceIpAddr>>,
+    /// Whether per‐interface IPv4 sysctl forwarding is enabled.
+    /// When applying with `Some(true)`, nmstate will enable forwarding.
+    /// When applying with `Some(false)`, nmstate will disable forwarding.
+    /// When applying with `None`, forwarding uses NetworkManager's current
+    /// or default setting ("auto"), which may result in NetworkManager
+    /// automatically enabling or disabling sysctl forwarding based on the
+    /// connections configuration (e.g., enabling it for "shared" mode).
```

Comment:
I would remove all these lines, except the first one. True and false are obvious, and None just have the same meaning than anywhere else: don't touch current value.

Reply:
This is top level API, it should contains implementation detail of NetworkManager.

No need to explains the detail,  meaning of `Some(true)`,  `Some(false)`  and `None` are obvious.

Reply:
@cathay4t , is there a typo,  it should or shoud not contain implementation details of NetworkManager?